flocker/node/agents/ebs.py

```diff
@@ -609,7 +609,10 @@ def compute_instance_id(self):
         """
         Look up the EC2 instance ID for this node.
         """
-        return get_instance_metadata()['instance-id'].decode("ascii")
+        instance_id = get_instance_metadata().get('instance-id', None)
+        if not instance_id:
```

Comment:
Oughtn't this be something more like `instance_id is None`?  Or a handler for `KeyError`?

Any idea if this case is actually possible or is it just that it looks possible due to symmetry with OpenStack?